sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -2133,7 +2133,7 @@ void CurlConnectionPool::MoveConnectionBackToPool(
 {
   auto code = static_cast<std::underlying_type<Http::HttpStatusCode>::type>(lastStatusCode);
   // laststatusCode = 0
-  if (code < 200 || code >= 300)
+  if ((code < 200 || code >= 300) && code != 404)
```

Comment:
Prefer using the symbolic name for the status code, rather than its numeric code.

```suggestion
  if ((code < 200 || code >= 300) && lastStatusCode != Http::HttpStatusCode::NotFound)
```

Reply:
Done

Reply:
THank you very much.